runtime/include/chpl-mem.h

```diff
@@ -198,8 +199,15 @@ static inline
 void chpl_mem_array_free(void* p,
                          size_t nmemb, size_t eltSize,
                          int32_t lineno, int32_t filename) {
-  if (!chpl_comm_regMemFree(p, nmemb * eltSize))
-    chpl_mem_free(p, lineno, filename);
+  const size_t size = nmemb * eltSize;
+
+  if (size >= chpl_comm_regMemAllocThreshold()
+      && size >= 2 * chpl_comm_regMemHeapPageSize()) {
+    if (chpl_comm_regMemFree(p, size))
```

Comment:
This seems shady to me. Why does `chpl_comm_regMemFree` have a return value? It seems like it's return value is whether memory can actually be registered with the NIC, but that seems like a bit of an abuse to me.

Reply:
5a1b0ec7 added a comment here to help explain, but also see the existing [block comment](https://github.com/chapel-lang/chapel/blob/master/runtime/include/chpl-comm.h#L246) which explains why `chpl_comm_regMemFree()` returns a "mine vs. not mine" indicator.